File: todoMeterRest/todometer/src/main/java/com/todometer/model/Project.java
package com.todometer.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonIgnoreType;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

/**
 * @Author Miguelet
 */
@Entity
@Table
@Data
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer projectId;
    @Column
    private String projectName;
    @Column
    private String projectDescription;
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "taskProjectId")
    @JsonIgnoreProperties("taskProjectId")
    private List<Task> tasks;
}

File: app/src/main/java/com/sergiobelda/androidtodometer/model/Project.kt
package com.sergiobelda.androidtodometer.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "project_table")
data class Project(
    var projectName: String,
    var projectDescription: String
) {
    @PrimaryKey(autoGenerate = true)
    var projectId: Int = 0

    override fun toString(): String {
        return projectName
    }
}